components/hab/src/command/pkg/binlink.rs

```diff
@@ -178,20 +193,57 @@ struct Binlink {
 
 #[cfg(not(target_os = "windows"))]
 impl Binlink {
-    pub fn new<T: AsRef<Path>>(src: T, dest_dir: T) -> Result<Self> {
+    pub fn new<T: AsRef<Path>>(
+        src: T,
+        dest_dir: T,
+        stub: T,
+        pkg_ident: &PackageIdent,
+    ) -> Result<Self> {
+        use std::os::unix::fs::OpenOptionsExt;
+
         let bin_name = match src.as_ref().file_name() {
             Some(name) => name,
             None => return Err(Error::CannotParseBinlinkSource(src.as_ref().to_path_buf())),
         };
 
-        Ok(Self {
-            dest: dest_dir.as_ref().join(bin_name),
-            src: src.as_ref().to_path_buf(),
-        })
+        let template = format!(
+            "\
+#!/bin/sh 
+if ! [ -x \"$(command -v hab)\" ]; then 
+    echo 'ERROR: The core/hab package needs to be binlinked' 
```

Comment:
Doesn't necessarily need to be binlinked, just on `$PATH`, right?

Reply:
Fixed